src/delay.rs

```diff
@@ -9,27 +9,18 @@
 
 /// Blocking delay traits
 pub mod blocking {
-    /// Millisecond delay
-    ///
-    /// `UXX` denotes the range type of the delay time. `UXX` can be `u8`, `u16`, etc. A single type can
-    /// implement this trait for different types of `UXX`.
-    pub trait DelayMs<UXX> {
-        /// Enumeration of `DelayMs` errors
-        type Error: core::fmt::Debug;
-
-        /// Pauses execution for `ms` milliseconds
-        fn delay_ms(&mut self, ms: UXX) -> Result<(), Self::Error>;
-    }
-
     /// Microsecond delay
     ///
-    /// `UXX` denotes the range type of the delay time. `UXX` can be `u8`, `u16`, etc. A single type can
-    /// implement this trait for different types of `UXX`.
-    pub trait DelayUs<UXX> {
-        /// Enumeration of `DelayMs` errors
+    pub trait DelayUs {
+        /// Enumeration of `DelayUs` errors
         type Error: core::fmt::Debug;
 
         /// Pauses execution for `us` microseconds
-        fn delay_us(&mut self, us: UXX) -> Result<(), Self::Error>;
+        fn delay_us(&mut self, us: u32) -> Result<(), Self::Error>;
+
+        /// Pauses execution for `ms` milliseconds
+        fn delay_ms(&mut self, ms: u32) -> Result<(), Self::Error> {
+            self.delay_us(ms * 1000)
```

Comment:
I'm not sure `delay_ms` is needed. 2^32 us is ~1.2h, doing blocking delays for that long should be super rare. 

And if you do such a long delay, the `ms*1000` here will overflow, so you can't anyway.

Reply:
It's not strictly needed, but as a convenience function it's nice. Always specifying `delay_us(154_000)` is a bit annoying.

Reply:
Then maybe `checked_mul` with an explicit panic could be used? If you're waiting in the ms range, the overhead should be too bad, but on the other hand silent overflows are bad.

Reply:
`delay_ms(u32)` imo implies that any u32 value works. It's surprising that it panics if it's too big, especially since the reason you may want to use it is for the increased range.

Reply:
This is a bit of a double edged nicity. I wanted to make migration simple so people can keep their `delay.delay_ms` code. We already implemented it this way in the riscv anyhow. I'm sure we should do a better job but I think that comes with the Time types.

Reply:
Applied solution from @adamgreig which avoids overflow

Reply:
With the overflow fix there's no longer any downside I guess! Nice :+1: